test/balancer_test.go

```diff
@@ -1128,7 +1133,7 @@ func (s) TestBalancerProducerHonorsContext(t *testing.T) {
 	ctxChan <- ctx
 
 	rpcErrChan := make(chan error)
-	balancer.Register(&producerTestBalancerBuilder{rpcErrChan: rpcErrChan, ctxChan: ctxChan})
+	balancer.Register(&producerTestBalancerBuilder{rpcErrChan: rpcErrChan, ctxChan: ctxChan, connect: false})
```

Comment:
I think it's a case-by-case basis.  It is a pretty important part of this test's behavior, so I think it's good to include it.  If it's some random config value that has an obvious/sensible default, then I think it would be fine to exclude.